### PR TITLE
Merge lyft cartography

### PR DESCRIPTION
**Pull request by sagarnavale**

### CODE REVIEW

**Review by ajinkya1986 — APPROVED** [review]
Looks good

File: cartography/intel/aws/elasticsearch.py [hunk 1]
```diff
@@ -322,7 +301,11 @@ def sync(
...
     update_tag: int, common_job_parameters: Dict,
 ) -> None:
-    tic = time.perf_counter()
+    for region in regions:
+        logger.info("Syncing Elasticsearch Service for region '%s' in account '%s'.", region, current_aws_account_id)
+        client = boto3_session.client('es', region_name=region, config=_get_botocore_config())
+        data = _get_es_domains(client)
+        _load_es_domains(neo4j_session, data, current_aws_account_id, update_tag)
 
     logger.info("Syncing Elasticsearch Service for account '%s', at %s.", current_aws_account_id, tic)
```
Comment (ganeshjagdale9545) [inline cartography/intel/aws/elasticsearch.py:310]: `tic` variable undefined

Comment (ganeshjagdale9545) [reply cartography/intel/aws/elasticsearch.py:310]: also `es_regions`

File: cartography/intel/aws/rds.py [hunk 2]
```diff
@@ -579,6 +528,100 @@ def load_rds_instances(
     _attach_clusters(neo4j_session, clusters, aws_update_tag)
 
 
+@timeit
+@aws_handle_regions
+def get_rds_snapshot_data(boto3_session: boto3.session.Session, region: str) -> List[Any]:
```
Comment (ganeshjagdale9545) [inline cartography/intel/aws/rds.py:533]: Method unused

File: cartography/intel/aws/rds.py [hunk 3]
```diff
@@ -766,6 +825,14 @@ def cleanup_rds_clusters(neo4j_session: neo4j.Session, common_job_parameters: Di
     run_cleanup_job('aws_import_rds_clusters_cleanup.json', neo4j_session, common_job_parameters)
 
 
+@timeit
+def cleanup_rds_snapshots(neo4j_session: neo4j.Session, common_job_parameters: Dict) -> None:
```
Comment (ganeshjagdale9545) [inline cartography/intel/aws/rds.py:829]: Method name defines a twice

File: cartography/intel/aws/rds.py [hunk 4]
```diff
@@ -860,10 +927,18 @@ def sync(
         neo4j_session, boto3_session, regions, current_aws_account_id, update_tag,
         common_job_parameters,
     )
-    sync_rds_reserved_db_instances(
+    sync_rds_snapshots(
```
Comment (ganeshjagdale9545) [inline cartography/intel/aws/rds.py:930]: Method called twice in `sync` function

File: cartography/intel/aws/rds.py [hunk 5]
```diff
@@ -860,10 +927,18 @@ def sync(
...
         neo4j_session, boto3_session, regions, current_aws_account_id, update_tag,
         common_job_parameters,
     )
+    merge_module_sync_metadata(
+        neo4j_session,
+        group_type='AWSAccount',
+        group_id=current_aws_account_id,
+        synced_type='RDSCluster',
+        update_tag=update_tag,
+        stat_handler=stat_handler,
```
Comment (ganeshjagdale9545) [inline cartography/intel/aws/rds.py:940]: `stat_handler` undefind

File: cartography/intel/aws/resourcegroupstaggingapi.py [hunk 6]
```diff
@@ -124,13 +125,18 @@ def get_short_id_from_lb2_arn(alb_arn: str) -> str:
     'iam:role': {'label': 'AWSRole', 'property': 'arn'},
     'iam:user': {'label': 'AWSUser', 'property': 'arn'},
     'kms:key': {'label': 'KMSKey', 'property': 'arn'},
+    'iam:group': {'label': 'AWSGroup', 'property': 'arn'},
+    'iam:role': {'label': 'AWSRole', 'property': 'arn'},
+    'iam:user': {'label': 'AWSUser', 'property': 'arn'},
```
Comment (ganeshjagdale9545) [inline cartography/intel/aws/resourcegroupstaggingapi.py:130]: `iam:group`, `iam:role`, `iam:user` are duplicate keys

File: cartography/intel/aws/resourcegroupstaggingapi.py [hunk 7]
```diff
@@ -124,13 +125,18 @@ def get_short_id_from_lb2_arn(alb_arn: str) -> str:
...
     'lambda:function': {'label': 'AWSLambda', 'property': 'id'},
     'lambda:layer': {'label': 'AWSLambdaLayer', 'property': 'id'},
     'lambda:event-source-mapping': {'label': 'AWSLambdaEventSourceMapping', 'property': 'id'},
     'redshift:cluster': {'label': 'RedshiftCluster', 'property': 'id'},
     'rds:cluster': {'label': 'RDSCluster', 'property': 'id'},
     'rds:db': {'label': 'RDSInstance', 'property': 'id'},
     'rds:subgrp': {'label': 'DBSubnetGroup', 'property': 'id'},
+    'rds:cluster': {'label': 'RDSCluster', 'property': 'id'},
```
Comment (ganeshjagdale9545) [inline cartography/intel/aws/resourcegroupstaggingapi.py:138]: same as above

File: cartography/intel/aws/resourcegroupstaggingapi.py [hunk 8]
```diff
@@ -245,22 +257,20 @@ def sync(
     common_job_parameters: Dict,
     tag_resource_type_mappings: Dict = TAG_RESOURCE_TYPE_MAPPINGS,
 ) -> None:
-    tic = time.perf_counter()
-
-    logger.info("Begin processing tags for account '%s', at %s.", current_aws_account_id, tic)
-
-    # Process each region in parallel.
-    with ThreadPoolExecutor(max_workers=len(regions)) as executor:
```
Comment (ganeshjagdale9545) [inline cartography/intel/aws/resourcegroupstaggingapi.py:253]: Don’t remove `ThreadPoolExecutor`  logic

File: cartography/intel/azure/cosmosdb.py [hunk 9]
```diff
@@ -399,7 +399,8 @@ def _load_cosmosdb_virtual_network_rules(
     if 'virtual_network_rules' in database_account and len(database_account['virtual_network_rules']) > 0:
         database_account_id = database_account['id']
         virtual_network_rules = database_account['virtual_network_rules']
-        virtual_network_rules['consolelink'] = azure_console_link.get_console_link(id=virtual_network_rules['id'], primary_ad_domain_name=common_job_parameters['Azure_Primary_AD_Domain_Name'])
+        for network_rule in virtual_network_rules:
+            network_rule['consolelink'] = azure_console_link.get_console_link(id=network_rule['id'], primary_ad_domain_name=common_job_parameters['Azure_Primary_AD_Domain_Name'])
```
Comment (ganeshjagdale9545) [inline cartography/intel/azure/cosmosdb.py:403]: Thanks for fixing this

File: cartography/intel/aws/elasticsearch.py (hunk 1 above)
Comment (sagarnavale) [reply cartography/intel/aws/elasticsearch.py:310]: tic and es_regions are defined

**Review by ganeshjagdale9545 — CHANGES_REQUESTED** [review]
few minor changes needed

File: cartography/intel/aws/rds.py (hunk 2 above)
Comment (sagarnavale) [reply cartography/intel/aws/rds.py:533]: removed it

File: cartography/intel/aws/rds.py (hunk 3 above)
Comment (sagarnavale) [reply cartography/intel/aws/rds.py:829]: one method is removed

File: cartography/intel/aws/rds.py (hunk 4 above)
Comment (sagarnavale) [reply cartography/intel/aws/rds.py:930]: one is removed

File: cartography/intel/aws/rds.py (hunk 5 above)
Comment (sagarnavale) [reply cartography/intel/aws/rds.py:940]: stat_handler is defined

File: cartography/intel/aws/resourcegroupstaggingapi.py (hunk 6 above)
Comment (sagarnavale) [reply cartography/intel/aws/resourcegroupstaggingapi.py:130]: removed it

File: cartography/intel/aws/resourcegroupstaggingapi.py (hunk 7 above)
Comment (sagarnavale) [reply cartography/intel/aws/resourcegroupstaggingapi.py:138]: removed it

**Pull request merged**